micro-rdk/src/esp32/board.rs

```diff
@@ -70,28 +70,25 @@ impl EspBoard {
         let (analogs, mut pins, i2c_confs) = {
             let analogs =
                 if let Ok(analogs) = cfg.get_attribute::<Vec<AnalogReaderConfig>>("analogs") {
-                    let analogs: Vec<AnalogReaderType<u16>> = analogs
+                    let analogs: Result<Vec<AnalogReaderType<u16>>, BoardError> = analogs
                         .iter()
-                        .filter_map(|v| {
-                            let adc1 = Arc::new(Mutex::new(
-                                AdcDriver::new(
-                                    unsafe { ADC1::new() },
-                                    &Config::new().calibration(true),
-                                )
-                                .ok()?,
-                            ));
-                            let chan: AnalogReaderType<u16> = match v.pin {
+                        .map(|v| {
+                            let adc1 = Arc::new(Mutex::new(AdcDriver::new(
+                                unsafe { ADC1::new() },
+                                &Config::new().calibration(true),
+                            )?));
+                            let chan: Result<AnalogReaderType<u16>, BoardError> = match v.pin {
                                 32 => {
                                     let p: AnalogReaderType<u16> =
                                         Arc::new(Mutex::new(Esp32AnalogReader::new(
                                             v.name.to_string(),
                                             AdcChannelDriver::<Atten11dB, _>::new(unsafe {
                                                 crate::esp32::esp_idf_svc::hal::gpio::Gpio32::new()
                                             })
-                                            .ok()?,
+                                            .map_err(BoardError::EspError)?,
```

Comment:
Originally, if a esp32analogreader failed to be created (esp32error) we just turned it into a none and ignored it. Do we want to return an error and fail the config now or ignore it like prev?

Reply:
It should return the error and fail to create the board